nexus_constructor/model/transformation.py

```diff
@@ -165,22 +166,23 @@ def as_dict(self) -> Dict[str, Any]:
         if isinstance(self.values, Dataset):
             if np.isscalar(self.values.values):
                 value = float(self.values.values)
-
+            return_dict = {
+                CommonKeys.NAME: self.name,
+                CommonKeys.TYPE: NodeType.DATASET,
+                CommonKeys.VALUES: value if value is not None else [],
+                CommonKeys.DATASET: {
+                    CommonKeys.TYPE: self.type,
+                    CommonKeys.SIZE: self.size,
+                },
+            }
+        elif isinstance(self.values, StreamGroup):
```

Comment:
Line 164: `as_dict()` method is called twice when we add two translations without kafkaStream. However if we add two translations both with kafkaStream, this method get called only once. Something fishy?